### ISSUE TITLE
TODO: Make rsc work in interactive Slurm jobs

### ISSUE THREAD

**Issue by HenrikBengtsson** [UCSF-CBI/rstudio-server-controller#14]
Adding here so I don't forget it.  Using:

```sh
$ srun --export=NONE --nodes=1 --cpus-per-task=4 --mem=32G --time=03:00:00 --pty $SHELL
srun: job 400020 queued and waiting for resources
srun: job 400020 has been allocated resources
...
```

I currently get:

```sh
[henrik@c4-n3:job=400020 rstudio-server-launcher]$ module load CBI r rstudio-server
[henrik@c4-n3:job=400020 rstudio-server-launcher]$ bin/rsc start
mkdir: cannot create directory ‘/.config’: Permission denied
mkdir: cannot create directory ‘/.config’: Permission denied
mkdir: cannot create directory ‘/.config’: Permission denied
mkdir: cannot create directory ‘/.config’: Permission denied
chmod: cannot access ‘/.config/rsc/run’: No such file or directory
chmod: cannot access ‘/.config/rsc/tmp’: No such file or directory
chmod: cannot access ‘/.config/rsc/var/lib/rstudio-server’: No such file or directory
chmod: cannot access ‘/.config/rsc//var/run/rstudio-server’: No such file or directory
bin/rsc: line 128: /.config/rsc/pid.lock: No such file or directory
/.config/rsc/pid.lock
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
ERROR: No such directory: '/.config/rsc' (working directory '/c4/home/henrik/repositories/UCSF-CBI/rstudio-server-launcher')
Traceback:
1: assert_dir_exists() on line #29 in bin/incl/asserts.sh
2: make_database_conf() on line #143 in bin/rsc
3: main() on line #553 in bin/rsc
Exiting (exit 1)
```

**Comment by HenrikBengtsson** [comment]
Ah... This is because `HOME` is not set;

```sh
[henrik@c4-n3:job=400020 rstudio-server-launcher]$ echo "HOME=$HOME" 
HOME=
```

**Comment by HenrikBengtsson** [comment]
@hgputnam, do you know why HOME is not set in interactive jobs?

**Comment by hgputnam** [comment]
If I launch srun without the --export=NONE, then $HOME is set. I think it is the --export flag...

**Comment by HenrikBengtsson** [comment]
I'm still sooo confused by that option. And, I thought the bash startup process sets HOME, but maybe that is not even performed here. Need to understand all this better in order to provide better docs.

**Comment by HenrikBengtsson** [comment]
I've "released" v0.1.1, which is merged in the main branch.  All development is done in the devel branch.  It's a first working prototype, but I need to do some more work on getting `R_LIBS_USER` correct and some cleanup on the output.  But here's a working example:

```sh
[alice@c4-log1 ~]$ srun --nodes=1 --cpus-per-task=4 --mem=32G --time=03:00:00 --pty $SHELL
...
[alice@c4-n3:job=400963 ~]$ module load rstudio-server-controller
[alice@c4-n3:job=400963 ~]$ rsc start --port=random
/c4/home/alice/.config/rsc/pid.lock
RStudio Server is available on <http://c4-n3:51751>.  If you're are running from a remote machine
without direct access to c4-n3, you can use SSH port forwarding to access the RStudio Server at
<http://127.0.0.1:8787> by running 'ssh -J alice@c4-log2.ucsf.edu -L 8787:c4-n3:51751 c4-n3' in
a second terminal.
```
